### ISSUE TITLE
implement missing/default members on trait doesnt include spaces if trait definition is via `include!(..)`'ed file or macro-generated

### ISSUE THREAD

**Issue by jhgg** [rust-analyzer/rust-analyzer#6588]
Minimal repro, have 2 files:

### a.rs
```rust
include!("a.rs");

struct Bar;

impl Foo for Bar {
   # <--- execute "implement missing members" here.
}
```

### b.rs
```rust
trait Foo {
    fn bar(&self, foo: Vec<u8>);
}
```

Output is:
```rust
impl Foo for Bar {
    fnbar(&self,foo:Vec<u8>) {
        todo!()
    }
}
```

However, if the trait is not `include!(...)`'ed, then it works fine! 

Ran into this issue, because tonic/prost generates rust files for protobufs, that you then use `include!(...)` to put the trait and struct definitions in the right place (can't just use regular `use` because of some weird limitations between how it generates output file names, and what cargo/rustc will accept.)

**Comment by Veykril** [comment]
Should be duplicate of #4665 and #2765 I believe. Macro generated syntax doesn't contain whitespace hence the issue due to the assist just copying the trait's functions and appending a body to them resulting in this output.

**Comment by lnicola** [comment]
Let's close this as a duplicate of #2765. @flodiebold please reopen if you think `include!` is different enough to warrant a separate issue.

**Comment by flodiebold** [comment]
I don't think `include` is different, but we should make sure that the fix also applies to code actions.

**Comment by jhgg** [comment]
It looks like #2765 and #4665 have been resolved by #8000 - however this issue still seems to remain. Same repro steps. Should we re-open?

**Comment by Veykril** [comment]
Ye this can be re-opened, the assist doesn't render its output from the HIR
